javatests/arcs/core/host/LifecycleParticles.kt

```diff
@@ -229,3 +229,34 @@ class ReadWriteAccessParticle : AbstractReadWriteAccessParticle() {
         }
     }
 }
+
+class PipelineProducerParticle : AbstractPipelineProducerParticle() {
+    override fun onFirstStart() {
+        handles.sngWrite.store(Value("sng"))
+        handles.colWrite.store(Value("col"))
+    }
+}
+
+class PipelineTransportParticle : AbstractPipelineTransportParticle() {
+    override fun onReady() {
+        handles.sngRead.fetch()!!.let {
```

Comment:
Optional: you can add `require` statements within these particles. That way, if later on the end result of the test fails (in the `pipeline` method), we'd have a better idea about where things went wrong.

Reply:
I didn't bother because (a) that's quite unlikely, and (b) if it does happen (bi) we should still have the line number in the stack trace and (bii) many, many other tests will also be broken :)

Reply:
Cool, that's a satisfying answer for me.